static/src/javascripts/projects/common/modules/commercial/contributions-service.js

```diff
@@ -169,16 +196,81 @@ const checkResponseOk = response => {
     );
 };
 
+// TODO: add this to the client library
+const getStickyBottomBanner = (payload: {}) => {
+    const URL = 'https://contributions.guardianapis.com/banner';
+    const json = JSON.stringify(payload);
+
+    return fetchJson(URL, {
+        method: 'post',
+        headers: { 'Content-Type': 'application/json' },
+        body: json,
+    });
+};
+
+export const fetchBannerData: () => Promise<?ServiceModule> = () => {
+    const payload = buildBannerPayload();
+
+    return getStickyBottomBanner(payload)
+        .then(json => {
+            if (!json.data) {
+                return null;
+            }
+
+            const { module } = json.data;
+
+            return module;
+        });
+};
+
+export const renderBanner: (?ServiceModule) => Promise<boolean> = (module) => {
+    if(!module) {
+        return Promise.resolve(false);
+    }
+
+    window.guardian.automat = {
+        react: React,
+        emotionCore,
+        emotionTheming,
+        emotion,
+    };
+
+    // $FlowFixMe
+    return import(/* webpackIgnore: true */ module.url)
+        .then(bannerModule => {
+            const Banner = bannerModule.Banner;
+            const props = { ...bannerModule.props };
```

Comment:
Is this the right place to be getting the props? I _think_ maybe this should be `module.props`?

Reply:
Yes I think you're right, that's a good spot!